src/UnivPoly.jl

```diff
@@ -0,0 +1,35 @@
+###############################################################################
+#
+#   Specialized methods for universal polynomials
+#
+###############################################################################
+
+# These methods speed up some computations with specific universal
+# polynomial rings which would otherwise be handled by more generic
+# code. This mainly concerns rings over the rationals.
+
+denominator(f::UniversalPolyRingElem{QQFieldElem}) = denominator(data(f))
+
+function +(p::Generic.UnivPoly{T}, n::ZZRingElem) where {T}
```

Comment:
Wouldn't all of these here make sense for, let's say, all "scalar" types that have ad-hoc operations?

In particular, I am thinking about `Integer` and `T` (the coeff type), but I guess those would belong in AA instead.

Reply:
Such methods already exist for `Integer` and `T` in AA. But not for `ZZRingElem` for obvious reasons. Which is why we add them here.